File: 101_200/167_two_sum_2_input_array_is_sorted/Solution.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int ri = numbers.length - 1;
        for (int i = 0; i < numbers.length ; i++) {
            int s = target - numbers[i];
            while (i < ri) {
                if (s == numbers[ri]) {
                    return new int[]{i+1, ri +1};
                } else if ( s < numbers[ri]) {
                    ri--;
                } else {
                    break;
                }
            }
        }
        return new int[] {};
    }
}

File: 101_200/167_two_sum_2_input_array_is_sorted/Solution.rs
impl Solution {
    pub fn two_sum(numbers: Vec<i32>, target: i32) -> Vec<i32> {
        let mut i = numbers.len() - 1;
        for (l, n) in numbers.iter().enumerate() {
            let s = target - n;
            while l < i {
                let pos:i32 = numbers[i];
                if pos == s {
                    return vec![l as i32 + 1, i as i32 + 1];
                } else if pos > s {
                    i -= 1;
                } else {
                    break;
                }
            }
        }
        return vec![];
    }
}